tracks/javascript/exercises/leap/mentoring.md

```diff
@@ -1,27 +1,67 @@
+The point of this exercise is to teach the student to use:
+- [Remainder (%)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Remainder_()), also known as modulo
+- [Logical operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators)
+- [Operator precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence)
+
 ### Reasonable solutions
 
 ```javascript
-export const isLeap = (year) => year % 4 === 0 && year % 100 !== 0 || year % 400 === 0;
+export const isLeap = (year) => year % 4 === 0 && (year % 100 !== 0 || year % 400 === 0);
 ```
+Variations check for truthy `year % m` and falsy `(!year % m)`.
 
 ```javascript
-export const isLeap = (year) => !(year % 4) && year % 100 || !(year % 400);
-```
+  function isDivisibleByFn(mod) {
+    return number => number % mod === 0;
+  }
+
+  const by4 = isDivisibleByFn(4);
+  const by100 = isDivisibleByFn(100);
+  const by400 = isDivisibleByFn(400);
+
+  export const isLeap = year => by4(year) && (!by10(year) || by400(year))
```

Comment:
The "reasonable solution" introduced in this pull request fails the exercise's test suite.

Reply:
Fixed. Don't know what happened there.